drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -0,0 +1,873 @@
+/*
+ * Copyright (c) 2020 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_i2c
+
+#include <drivers/i2c.h>
+#include <errno.h>
+#include <logging/log.h>
+		LOG_MODULE_REGISTER(i2c_ite_it8xxx2);
+#include "i2c-priv.h"
+#include <soc.h>
+#include <sys/util.h>
+
+#define DEV_CFG(dev) \
+		((const struct i2c_it8xxx2_config * const)(dev)->config)
+#define DEV_DATA(dev) \
+		((struct i2c_it8xxx2_data * const)(dev)->data)
+
+#define I2C_STANDARD_PORT_COUNT 3
+/* Default PLL frequency. */
+#define PLL_CLOCK 48000000
+
+struct i2c_it8xxx2_config {
+	uint8_t *base;
+	void (*irq_config_func)(void);
+	uint32_t bitrate;
+	uint32_t i2c_irq_base;
+	int port;
+};
+
+enum i2c_ch_status {
+	I2C_CH_NORMAL = 0,
+	I2C_CH_REPEAT_START,
+	I2C_CH_WAIT_READ,
+	I2C_CH_WAIT_NEXT_XFER,
+};
+
+struct i2c_it8xxx2_data {
+	struct k_sem device_sync_sem;
+	uint16_t addr_16bit;
+	struct i2c_msg *msgs;
+	int err;			 /* Error code, if any */
+	int widx;			 /* Index into output data */
```

Comment:
If these indexes are non-negative, use an unsigned type.  If there's hardware constraints on the maximum offset use a smaller type; otherwise use `size_t`.

Reply:
Done